quickwit/quickwit-metastore/src/metastore/mod.rs

```diff
@@ -496,6 +508,16 @@ impl ListSplitsQuery {
         self.create_timestamp.start = Bound::Excluded(v);
         self
     }
+
+    /// Set the maturity filter to match splits that are mature or immature
+    /// at a given datetime.
+    pub fn is_mature(mut self, mature: bool, evaluation_datetime: OffsetDateTime) -> Self {
```

Comment:
Second nitpick...

`is_xxxx` are usually function that accessor returning a `bool`. Here it is tricky because you are building a predicate, which makes it sensible to use this vocab.

`filter_xxx` is ironically even worse, because `filter` itself is ambiguous. Some program call filter stuff that `filter out`, some that `retain`.

I would suggest two methods (we don't use it dynamically I think) called 

`fn retain_mature(now: OffsetDateTime)`
`fn retain_immature(now: OffsetDateTime)`

Reply:
good idea, `retain` really improves the API.